src/ApacheOrcDotNet.OptimizedReader/Buffers/BaseColumnBuffer.cs

```diff
@@ -101,7 +81,90 @@ private protected void ReadBooleanStream(StreamDetail stream, ReadOnlySpan<byte>
                 return;
 
             var numSkipped = 0;
-            var bufferReader = new BufferReader(ResizeBuffer(stream, buffer, length));
+            var bufferReader = new BufferReader(GetDataStream(stream, buffer, length));
+            var numOfTotalBitsToSkip = stream.Positions.ValuesToSkip * 8 + stream.Positions.RemainingBits;
+            var numOfBytesToSkip = numOfTotalBitsToSkip / 8;
+            while (!bufferReader.Complete)
+            {
+                var numByteValuesRead = OptimizedByteRLE.ReadValues(ref bufferReader, _boolStreamBuffer);
+
+                for (int idx = 0; idx < numByteValuesRead; idx++)
+                {
+                    if (numSkipped++ < numOfBytesToSkip)
```

Comment:
This end up being the simplest way since the total `numOfBytesToSkip` can be distributed through different RLEs, forcing us to keep track of how many were already skipped.

Reply:
Yeah, that makes sense.